### PR TITLE
package discovery-engine as a systemd service

### PR DESCRIPTION
**Pull request by Ankurk99**
Signed-off-by: Ankur Kothiwal <ankur.kothiwal@accuknox.com>

### CODE REVIEW

File: src/packaging/discovery-engine.service [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[Unit]
+Description=knoxAutoPolicy
+
+[Service]
+User=root
```
Comment (nyrahul) [inline src/packaging/discovery-engine.service]: Do we really need root user?

**Review by nyrahul — CHANGES_REQUESTED** [review]
Mostly LGTM, except that the `user=root` is enabled for knoxAutoPolicy.

File: src/packaging/discovery-engine.service (hunk 1 above)
Comment (seswarrajan) [reply src/packaging/discovery-engine.service]: We do not need this to run as super user/root user.

File: src/.goreleaser.yml [hunk 2]
```diff
@@ -1,27 +1,71 @@
...
-  - env:
-      - CGO_ENABLED=1
+  - binary: "opt/knoxAutoPolicy/src/knoxAutoPolicy"
+    id: knoxAutoPolicy
     goos:
       - linux
         #- windows
         #- darwin
     goarch:
       - amd64
```
Comment (seswarrajan) [inline src/.goreleaser.yml:18]: Kindly include **arm64** as well

File: src/packaging/discovery-engine.service (hunk 1 above)
Comment (Ankurk99) [reply src/packaging/discovery-engine.service]: Not sure if we need to run discovery by root. But from here https://stackoverflow.com/a/53537418, it seems the service is by default started in root mode. I think we can remove it then

File: src/packaging/post-install.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+set -e
+
+/bin/systemctl daemon-reload
+/bin/systemctl start knoxAutoPolicy.service
```
Comment (seswarrajan) [inline src/packaging/post-install.sh:4]: Also if possible/required, add below line 

/bin/systemctl enable knoxAutoPolicy.service

**Review by seswarrajan — COMMENTED** [review]
Kindly address review comments if possible or it can be handled in later release.

File: src/packaging/post-install.sh (hunk 3 above)
Comment (Ankurk99) [reply src/packaging/post-install.sh:4]: Enabling the service will automatically start the discovery-engine on each system startup, I think that might not be required by the user. Wdyt?

File: src/.goreleaser.yml (hunk 2 above)
Comment (Ankurk99) [reply src/.goreleaser.yml:18]: I think we can add it once we support KubeArmor too. :grin:

File: src/packaging/post-install.sh (hunk 3 above)
Comment (nyrahul) [reply src/packaging/post-install.sh:4]: Yeah, lets not start by default.

**Review by nyrahul — APPROVED** [review]
LGTM

**Pull request merged**